### PR TITLE
BROS-38 Add timeout cancellation fro pending requests

### PR DESCRIPTION
**Pull request by bapierre**
Merge this.

### CODE REVIEW

File: analytics/src/main/java/brocoeur/example/analytics/service/ServiceRequestStatusStatus.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+public enum ServiceRequestStatusStatus {
+
+    IN_PROGRESS("IN_PROGRESS"),
+    TODO("TODO"),
+    DONE_WIN("DONE_WIN"),
+    DONE_LOSS("DONE_LOSS"),
+    REJECTED("REJECTED");
+
+    public final String label;
```
Comment (gaugerard) [inline analytics/src/main/java/brocoeur/example/analytics/service/ServiceRequestStatusStatus.java:11]: private with getter().

File: analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java [hunk 2]
```diff
@@ -34,7 +36,20 @@ public class PokerScheduledTask {
...
         LOGGER.info("Poker scheduled task started at : {}", dateFormat.format(new Date()));
-        final List<ServiceRequestStatus> listServiceRequestStatus = serviceRequestStatusRepository.findAllByStrategyAndStatus(POKER_RANDOM.toString(), "TODO").collectList().block();
+        final List<ServiceRequestStatus> listServiceRequestStatus = serviceRequestStatusService.findAllServiceRequestStatusByStrategyAndStatus(POKER_RANDOM.toString(), TODO);
         LOGGER.info("List of poker request : {}", listServiceRequestStatus);
     }
+
+    @Scheduled(fixedRate = 3000)
+    public void rejectBlockedRequests() {
+        LOGGER.info("Cleaning scheduled task started at : {}", dateFormat.format(new Date()));
+        List<ServiceRequestStatus> pendingRequests = serviceRequestStatusService.findAllServiceRequestStatusByStatus(IN_PROGRESS);
```
Comment (gaugerard) [inline analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java:46]: final keyword

File: analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java [hunk 3]
```diff
@@ -34,7 +36,20 @@ public class PokerScheduledTask {
...
+        for(ServiceRequestStatus request : pendingRequests){
+            if(request.getInsertionTimeMilliSecond() < (int)new Date().getTime() - 5000){
```
Comment (gaugerard) [inline analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java:48]: Need to be configurable.

File: analytics/src/test/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTaskTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                124,
+                IN_PROGRESS.label,
+                100,
+                8,
+                ROULETTE_SAFE.name(),
+                (int)new Date().getTime() - 500,
+                0
+        );
+
+        when(serviceRequestStatusServiceMock.findAllServiceRequestStatusByStatus(IN_PROGRESS)).thenReturn(Arrays.asList(blockedRequest,secondRequest));
```
Comment (gaugerard) [inline analytics/src/test/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTaskTest.java:76]: You can uses List.of(...)

File: analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java [hunk 5]
```diff
@@ -34,7 +36,20 @@ public class PokerScheduledTask {
     @Scheduled(fixedRate = 30000)
     public void executeScheduledTask() {
         LOGGER.info("Poker scheduled task started at : {}", dateFormat.format(new Date()));
-        final List<ServiceRequestStatus> listServiceRequestStatus = serviceRequestStatusRepository.findAllByStrategyAndStatus(POKER_RANDOM.toString(), "TODO").collectList().block();
+        final List<ServiceRequestStatus> listServiceRequestStatus = serviceRequestStatusService.findAllServiceRequestStatusByStrategyAndStatus(POKER_RANDOM.toString(), TODO);
         LOGGER.info("List of poker request : {}", listServiceRequestStatus);
     }
+
```
Comment (gaugerard) [inline analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java:42]: Add javadoc explaining the goal of this Scheduled task.

**Review by gaugerard — CHANGES_REQUESTED** [review]

File: analytics/src/main/java/brocoeur/example/analytics/service/ServiceRequestStatusStatus.java (hunk 1 above)
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/service/ServiceRequestStatusStatus.java:11]: Added Lombok @Getter

File: analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java (hunk 3 above)
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java:48]: To be done in a future card

File: analytics/src/test/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTaskTest.java (hunk 4 above)
Comment (bapierre) [reply analytics/src/test/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTaskTest.java:76]: Changed

File: analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java (hunk 5 above)
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java:42]: Added

File: analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java (hunk 2 above)
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/service/scheduler/PokerScheduledTask.java]: added

**Review by gaugerard — APPROVED** [review]

**Pull request merged**